rest/handler.go

```diff
@@ -283,27 +283,51 @@ func (h *handler) logStatus(status int, message string) {
 	h.logDuration(false) // don't track actual time
 }
 
-func (h *handler) checkAuth(context *db.DatabaseContext) error {
-	h.user = nil
-	if context == nil {
-		return nil
+func (h *handler) statAuthSuccess() {
+	if h.db == nil || h.db.DatabaseContext == nil || h.db.DatabaseContext.DbStats == nil {
+		return
+	}
+	h.db.DatabaseContext.DbStats.StatsSecurity().Add(base.StatKeyAuthSuccessCount, 1)
+}
+
+func (h *handler) statAuthFailed() {
+	if h.db == nil || h.db.DatabaseContext == nil || h.db.DatabaseContext.DbStats == nil {
+		return
 	}
+	h.db.DatabaseContext.DbStats.StatsSecurity().Add(base.StatKeyAuthFailedCount, 1)
+}
+
+func (h *handler) checkAuth(context *db.DatabaseContext) (err error) {
+
+	// Record stats
+	defer func() {
+		if err != nil {
+			h.statAuthFailed()
+		} else {
+			h.statAuthSuccess()
+		}
+	}()
```

Comment:
If you move this down below the context==nil check, it can just be:
```
	defer func() {
		if err != nil {
			h.db.DatabaseContext.DbStats.StatsSecurity().Add(base.StatKeyAuthFailedCount, 1)
		} else {
			h.db.DatabaseContext.DbStats.StatsSecurity().Add(base.StatKeyAuthSuccessCount, 1)
		}
	}()
```

Reply:
I originally had something like that, but then found it confused to have:

- code
- defer function definition
- defer function definition
- code

and thought it was slightly clearer to have all the defers stacked up at the top of the function before any "real code" was defined.  

I agree it's a bit clunky in it's current form though, so I'll change it back.